gateways/adyen/request_builder.go

```diff
@@ -152,6 +145,48 @@ func addPaymentSpecificFields(authRequest *sleet.AuthorizationRequest, request *
 	}
 }
 
+// addBillingAddress adds the billing address to the Ayden Payment request if available
+func addBillingAddress(authRequest *sleet.AuthorizationRequest, request *checkout.PaymentRequest) {
+	if authRequest.BillingAddress != nil {
+		request.BillingAddress = &checkout.Address{
+			City:              common.SafeStr(authRequest.BillingAddress.Locality),
+			Country:           common.SafeStr(authRequest.BillingAddress.CountryCode),
+			HouseNumberOrName: common.SafeStr(authRequest.BillingAddress.StreetAddress2),
+			PostalCode:        common.SafeStr(authRequest.BillingAddress.PostalCode),
+			StateOrProvince:   common.SafeStr(authRequest.BillingAddress.RegionCode),
+			Street:            common.SafeStr(authRequest.BillingAddress.StreetAddress1),
+		}
+	}
+}
+
+// addShopperIP adds the shoppers IP to the Ayden Payment request if available
+func addShopperIP(authRequest *sleet.AuthorizationRequest, request *checkout.PaymentRequest) {
+	if authRequest.ShopperIP != nil {
+		request.ShopperIP = common.SafeStr(authRequest.ShopperIP)
+	}
+}
+
+// addShopperEmail adds the shoppers email to the Ayden Payment request if available
+func addShopperEmail(authRequest *sleet.AuthorizationRequest, request *checkout.PaymentRequest) {
+	if authRequest.ShopperEmail != nil {
+		request.ShopperEmail = common.SafeStr(authRequest.ShopperEmail)
+	}
+}
+
+// addShippingAddress adds the shipping address to the Ayden Payment request if available
+func addShippingAddress(authRequest *sleet.AuthorizationRequest, request *checkout.PaymentRequest) {
+	if authRequest.ShippingAddress != nil {
+		request.DeliveryAddress = &checkout.Address{
+			City:              common.SafeStr(authRequest.ShippingAddress.Locality),
+			Country:           common.SafeStr(authRequest.ShippingAddress.CountryCode),
+			HouseNumberOrName: common.SafeStr(authRequest.ShippingAddress.StreetAddress2),
```

Comment:
I copied what we did for billing address but looking at ayden docs maybe we are using these fields incorrectly? Let me know thoughts https://docs.adyen.com/api-explorer/#/CheckoutService/latest/post/payments__reqParam_deliveryAddress

Reply:
Yeah weird it doesn't look like we are using `Street` and `HouseNumberOrName` correctly. @nirajjayant do you know why it was implemented this way?

Reply:
Let's just keep it consistent for now and then add a task to the backlog to evaluate changing how we do this.

Reply:
Sorry maybe I am not seeing something obvious - is StreetAddress2 not typical for HouseNumberOrName?

Reply:
According to adyen docs ->
123 Street Name
Mountain View, CA 94043

we would fill in as
HouseNumberOrName: 123
Street: Street Name
City: Mountain View
StateOrProvince: CA
PostalCode: 94043